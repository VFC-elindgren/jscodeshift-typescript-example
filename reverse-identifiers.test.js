'use strict';

const fs = require('fs');
const jscodeshift = require('jscodeshift');
const path = require('path');
const transform = require('./reverse-identifiers');

it('transforms correctly', () => {
  const inputPath = path.resolve(__dirname, 'reverse-identifiers.input.ts');
  const file = {
    path: inputPath,
    source: fs.readFileSync(inputPath, 'utf8'),
  }
  const output = transform(file, {
    jscodeshift,
    stats: () => {},
  });

  expect(output).toMatchSnapshot();
});
