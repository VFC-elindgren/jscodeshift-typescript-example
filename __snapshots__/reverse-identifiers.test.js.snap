// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transforms correctly 1`] = `
"type yarrAemaNrOemaN = string | string[];

function emaNetaerc(eman: yarrAemaNrOemaN) {
    if (typeof eman === \\"string\\") {
        return eman;
    }
    else {
        return eman.nioj(\\" \\");
    }
}

var egasseMgniteerg = \`Greetings, \${ emaNetaerc([\\"Sam\\", \\"Smith\\"]) }\`;
trela(egasseMgniteerg);
"
`;
